cmd/samples/recipes/searchattributes/searchattributes_workflow.go

```diff
@@ -0,0 +1,175 @@
+		"CustomKeywordField":  "Update1",
+		"CustomBoolField":     true,
+		"CustomDoubleField":   3.14,
+		"CustomDatetimeField": time.Date(2019, 1, 1, 0, 0, 0, 0, time.Local),
+		"CustomStringField":   "String field is for text. When query, it will be tokenized for partial match. StringTypeField cannot be used in Order By",
+	}
+	workflow.UpsertSearchAttributes(ctx, attributes)
+
+	// print current search attributes
+	info = workflow.GetInfo(ctx)
+	err = printSearchAttributes(info.SearchAttributes, logger)
+	if err != nil {
+		return err
+	}
+
+	// update search attributes again
+	attributes = map[string]interface{}{
+		"CustomKeywordField": "Update2",
+	}
+	workflow.UpsertSearchAttributes(ctx, attributes)
+
+	// print current search attributes
+	info = workflow.GetInfo(ctx)
+	err = printSearchAttributes(info.SearchAttributes, logger)
+	if err != nil {
+		return err
+	}
+
+	workflow.Sleep(ctx, 2*time.Second) // wait update reflected on ElasticSearch
+
+	// list workflow
+	ao := workflow.ActivityOptions{
+		ScheduleToStartTimeout: 2 * time.Minute,
+		StartToCloseTimeout:    2 * time.Minute,
+		HeartbeatTimeout:       time.Second * 20,
+	}
+	ctx = workflow.WithActivityOptions(ctx, ao)
+	query := "CustomIntField=2 and CustomKeywordField='Update2' order by CustomDatetimeField DESC"
+	var listResults []*shared.WorkflowExecutionInfo
+	err = workflow.ExecuteActivity(ctx, listExecutions, query).Get(ctx, &listResults)
+	if err != nil {
+		logger.Error("Failed to list workflow executions.", zap.Error(err))
+		return err
+	}
+
+	logger.Info("Workflow completed.", zap.String("Execution", listResults[0].String()))
+
+	return nil
+}
+
+func printSearchAttributes(searchAttributes *shared.SearchAttributes, logger *zap.Logger) error {
+	buf := new(bytes.Buffer)
+	for k, v := range searchAttributes.IndexedFields {
+		var currentVal interface{}
+		err := client.NewValue(v).Get(&currentVal)
+		if err != nil {
+			logger.Error(fmt.Sprintf("Get search attribute for key %s failed", k), zap.Error(err))
+			return err
+		}
+		fmt.Fprintf(buf, "%s=%v\n", k, currentVal)
+	}
+	logger.Info(fmt.Sprintf("Current Search Attributes: \n%s", buf.String()))
+	return nil
+}
+
+func listExecutions(ctx context.Context, query string) ([]*shared.WorkflowExecutionInfo, error) {
+	logger := activity.GetLogger(ctx)
+	logger.Info("List executions.", zap.String("Query", query))
+
+	cadenceClient, err := getCadenceClientFromContext(ctx)
+	if err != nil {
+		logger.Error("Error when get cadence client")
+		return nil, err
+	}
+
+	var executions []*shared.WorkflowExecutionInfo
+	var nextPageToken []byte
+	for hasMore := true; hasMore; hasMore = len(nextPageToken) > 0 {
+		resp, err := cadenceClient.ListWorkflow(ctx, &shared.ListWorkflowExecutionsRequest{
+			Domain:        common.StringPtr(DomainName),
+			PageSize:      common.Int32Ptr(10),
+			NextPageToken: nextPageToken,
+			Query:         common.StringPtr(query),
```

Comment:
Why not just `&query`? Do we use a helper function like this in the Cadence code as well?

Reply:
Yes we do use same conversion in Cadence repos, and I am following the convention. 
I think it just try to make code readable to avoid defining local `var` then use `&var`, personally I don't have preference on this.